src/pkt_checker.rs

```diff
@@ -0,0 +1,206 @@
+use std::{
+    collections::{HashMap, LinkedList},
+    sync::{atomic::{AtomicBool, Ordering}, mpsc::Sender, Arc, Mutex, RwLock},
+};
+
+use crate::{
+    op_ctx::ReadOpCtx,
+    recv_pkt_map::RecvPktMap,
+    responser::{RespAckCommand, RespCommand},
+    types::{Msn, Psn},
+    Error,
+};
+
+use log::{error, info};
+
+#[derive(Debug)]
+pub(crate) struct PacketChecker {
+    thread: Option<std::thread::JoinHandle<()>>,
+    stop_flag: Arc<AtomicBool>,
+}
+
+impl PacketChecker {
+    pub(crate) fn new(
+        send_queue: Sender<RespCommand>,
+        recv_pkt_map: Arc<RwLock<HashMap<Msn, Arc<Mutex<RecvPktMap>>>>>,
+        read_op_ctx_map: Arc<RwLock<HashMap<Msn, ReadOpCtx>>>,
+    ) -> Self {
+        let ctx = PacketCheckerContext {
+            send_queue,
+            recv_pkt_map,
+            read_op_ctx_map,
+        };
+        let stop_flag = Arc::new(AtomicBool::new(false));
+        let thread_stop_flag = Arc::clone(&stop_flag);
+        let thread = std::thread::spawn(move || {
```

Comment:
Why are you creating an thread here? I think a task is enough if you're using an multi-thread async runtime lib.